File: app/src/main/java/doit/study/droid/data/source/local/QuizDatabase.java
package doit.study.droid.data.source.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import doit.study.droid.data.source.local.entities.QuestionEntity;
import doit.study.droid.data.source.local.entities.QuestionTagJoin;
import doit.study.droid.data.source.local.entities.StatisticEntity;
import doit.study.droid.data.source.local.entities.TagEntity;


@Database(entities = {QuestionEntity.class, TagEntity.class, StatisticEntity.class, QuestionTagJoin.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class QuizDatabase extends RoomDatabase {
    public abstract QuestionDao getQuestionDao();
    public abstract TagDao getTagDao();
    public abstract StatisticDao getStatisticsDao();
    public abstract QuizDao getQuizDao();
}

File: app/src/main/java/doit/study/droid/data/local/QuizDatabase.kt
package doit.study.droid.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import doit.study.droid.data.local.dao.QuestionDao
import doit.study.droid.data.local.dao.TagDao
import doit.study.droid.data.local.entity.Question
import doit.study.droid.data.local.entity.QuestionTagJoin
import doit.study.droid.data.local.entity.Tag

@Database(entities = [Question::class, Tag::class, QuestionTagJoin::class], version = 2)
@TypeConverters(Converters::class)
abstract class QuizDatabase : RoomDatabase() {
    abstract fun questionDao(): QuestionDao
    abstract fun tagDao(): TagDao
}
